src/components/functional_module/include/functional_module/plugin_manager.h

```diff
@@ -38,12 +38,20 @@
 #include "functional_module/generic_module.h"
 #include "application_manager/service.h"
 #include "application_manager/message.h"
+#include "protocol_handler/remote_services_message_observer.h"
+#include "connection_handler/remote_services_observer.h"
 #include "utils/singleton.h"
 
+namespace protocol_handler {
+  class ProtocolHandler;
+}  // namespace protocol_handler
+
 namespace functional_modules {
 
 class PluginManager : public utils::Singleton<PluginManager>,
-  public ModuleObserver {
+  public ModuleObserver,
+  public protocol_handler::RemoteServicesMessageObserver,
```

Comment:
I disagree strongly. It's new dependency. Bottom layers should not know about top layers. This looks like Chain of Responsibility pattern when components is responsible for making decision to skip or handle message. I think connection_handler should be changed to use few observers simultaneously but I'm not sure if we have to do this (I mean scope of project).